pkg/psmdb/mongos.go

```diff
@@ -0,0 +1,361 @@
+		cfgInstanses = append(cfgInstanses,
+			fmt.Sprintf("%s-%s-%d.%s-%s.%s.svc.cluster.local:%d",
+				cr.Name, cfgRs.Name, i, cr.Name, cfgRs.Name, cr.Namespace, msSpec.Port))
+	}
+
+	configDB := fmt.Sprintf("%s/%s", cfgRs.Name, strings.Join(cfgInstanses, ","))
+	args := []string{
+		"mongos",
+		"--bind_ip_all",
+		"--port=" + strconv.Itoa(int(msSpec.Port)),
+		"--sslAllowInvalidCertificates",
+		"--configdb",
+		configDB,
+	}
+
+	if cr.Spec.UnsafeConf {
+		args = append(args,
+			"--clusterAuthMode=keyFile",
+			"--keyFile="+mongodSecretsDir+"/mongodb-key",
+		)
+	} else {
+		args = append(args,
+			"--sslMode=preferSSL",
+			"--clusterAuthMode=x509",
+		)
+	}
+
+	if mdSpec.Security != nil && mdSpec.Security.RedactClientLogData {
+		args = append(args, "--redactClientLogData")
+	}
+
+	if msSpec.SetParameter != nil {
+		if msSpec.SetParameter.CursorTimeoutMillis > 0 {
+			args = append(args,
+				"--setParameter",
+				"cursorTimeoutMillis="+strconv.Itoa(msSpec.SetParameter.CursorTimeoutMillis),
+			)
+		}
+	}
+
+	if msSpec.AuditLog != nil && msSpec.AuditLog.Destination == api.AuditLogDestinationFile {
+		if msSpec.AuditLog.Filter == "" {
+			msSpec.AuditLog.Filter = "{}"
+		}
+		args = append(args,
+			"--auditDestination=file",
+			"--auditFilter="+msSpec.AuditLog.Filter,
+			"--auditFormat="+string(msSpec.AuditLog.Format),
+		)
+		switch msSpec.AuditLog.Format {
+		case api.AuditLogFormatBSON:
+			args = append(args, "--auditPath="+MongodContainerDataDir+"/auditLog.bson")
+		default:
+			args = append(args, "--auditPath="+MongodContainerDataDir+"/auditLog.json")
+		}
+	}
+
+	return args
+}
+
+func volumes(cr *api.PerconaServerMongoDB) []corev1.Volume {
+	fvar := false
+	t := true
+	volumes := []corev1.Volume{
+		{
+			Name: InternalKey(cr),
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					DefaultMode: &secretFileMode,
+					SecretName:  InternalKey(cr),
+					Optional:    &fvar,
+				},
+			},
+		},
+		{
+			Name: "ssl",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName:  cr.Spec.Secrets.SSL,
+					Optional:    &cr.Spec.UnsafeConf,
+					DefaultMode: &secretFileMode,
+				},
+			},
+		},
+		{
+			Name: "ssl-internal",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName:  cr.Spec.Secrets.SSLInternal,
+					Optional:    &t,
+					DefaultMode: &secretFileMode,
+				},
+			},
+		},
+		{
+			Name: MongodDataVolClaimName,
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
+	}
+
+	if *cr.Spec.Mongod.Security.EnableEncryption {
+		volumes = append(volumes,
+			corev1.Volume{
+				Name: cr.Spec.Mongod.Security.EncryptionKeySecret,
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						DefaultMode: &secretFileMode,
+						SecretName:  cr.Spec.Mongod.Security.EncryptionKeySecret,
+						Optional:    &fvar,
+					},
+				},
+			},
+		)
+	}
```

Comment:
why it is needed?

Reply:
Do not needed
https://github.com/percona/percona-server-mongodb-operator/commit/be7bdefaa2b6f362495a8a77e1b79e5e50ad2560